code/go/0chain.net/chaincore/block/entity_test.go

```diff
@@ -0,0 +1,3314 @@
+		blockState            int8
+		isNotarized           bool
+		verificationStatus    int
+		RunningTxnCount       int64
+		UniqueBlockExtensions map[string]bool
+		MagicBlock            *MagicBlock
+	}
+	type args struct {
+		prevBlock *Block
+	}
+	tests := []struct {
+		name      string
+		fields    fields
+		args      args
+		debug     bool
+		want      *Block
+		wantPanic bool
+	}{
+		{
+			name: "OK",
+			fields: fields{
+				UnverifiedBlockBody:   b.UnverifiedBlockBody,
+				VerificationTickets:   b.VerificationTickets,
+				HashIDField:           b.HashIDField,
+				Signature:             b.Signature,
+				ChainID:               b.ChainID,
+				ChainWeight:           b.ChainWeight,
+				RoundRank:             b.RoundRank,
+				PrevBlock:             b.PrevBlock,
+				TxnsMap:               b.TxnsMap,
+				ClientState:           b.ClientState,
+				stateStatus:           b.stateStatus,
+				blockState:            b.blockState,
+				isNotarized:           b.isNotarized,
+				verificationStatus:    b.verificationStatus,
+				RunningTxnCount:       b.RunningTxnCount,
+				UniqueBlockExtensions: b.UniqueBlockExtensions,
+				MagicBlock:            b.MagicBlock,
+			},
+			args: args{prevBlock: prevB},
+			want: func() *Block {
+				b := NewBlock("", 1)
+				pndb := util.NewMemoryNodeDB()
+				rootHash := prevB.ClientStateHash
+				b.CreateState(pndb)
+				b.ClientState.SetRoot(rootHash)
+
+				return b
+			}(),
+		},
+		{
+			name:      "Debug_PANIC",
+			debug:     true,
+			wantPanic: true,
+		},
+		{
+			name: "Non_Nil_Client_State",
+			fields: fields{
+				UnverifiedBlockBody:   b.UnverifiedBlockBody,
+				VerificationTickets:   b.VerificationTickets,
+				HashIDField:           b.HashIDField,
+				Signature:             b.Signature,
+				ChainID:               b.ChainID,
+				ChainWeight:           b.ChainWeight,
+				RoundRank:             b.RoundRank,
+				PrevBlock:             b.PrevBlock,
+				TxnsMap:               b.TxnsMap,
+				ClientState:           b.ClientState,
+				stateStatus:           b.stateStatus,
+				blockState:            b.blockState,
+				isNotarized:           b.isNotarized,
+				verificationStatus:    b.verificationStatus,
+				RunningTxnCount:       b.RunningTxnCount,
+				UniqueBlockExtensions: b.UniqueBlockExtensions,
+				MagicBlock:            b.MagicBlock,
+			},
+			args: args{
+				prevBlock: func() *Block {
+					prevB := NewBlock("", 0)
+					prevB.ClientState = cs
+					return prevB
+				}(),
+			},
+
+			want: func() *Block {
+				b := NewBlock("", 1)
+				pndb := cs.GetNodeDB()
+				rootHash := prevB.ClientStateHash
+				b.CreateState(pndb)
+				b.ClientState.SetRoot(rootHash)
+
+				return b
+			}(),
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			defer func() {
+				got := recover()
+				if (got != nil) != tt.wantPanic {
+					t.Errorf("SetStateDB() want panic  = %v, but got = %v", tt.wantPanic, got)
+				}
+			}()
+
+			if tt.debug {
+				state.SetDebugLevel(1)
```

Comment:
I've split it up into two tests and duplicated.
Also, I've marked these tests as skipped because we need to protect `state.debugState` against concurrent access.

Reply:
No, don't skip. As long as we call SetDebug only once there is no problem.

Reply:
I've removed `t.Skip`